go/pkg/cas/client.go

```diff
@@ -192,15 +231,24 @@ func NewClientWithConfig(ctx context.Context, conn *grpc.ClientConn, instanceNam
 // creating a real gRPC connection. This function exists purely to aid testing,
 // and is tightly coupled with NewClientWithConfig.
 func (c *Client) init() {
-	c.semBatchUpdateBlobs = semaphore.NewWeighted(int64(c.BatchUpdateBlobsConcurrency))
 	c.semFileIO = semaphore.NewWeighted(int64(c.FSConcurrency))
 	c.fileIOBufs.New = func() interface{} {
 		return make([]byte, c.FileIOSize)
 	}
+
+	c.FindMissingBlobs.init()
+	c.BatchUpdateBlobs.init()
 }
 
-func (c *Client) withRetries(ctx context.Context, f func() error) error {
-	return retry.WithPolicy(ctx, retry.TransientOnly, c.RetryPolicy, f)
+// unaryRPC calls f with retries, and with per-RPC timeouts.
+// Does not limit concurrency.
+// It is useful when f calls an unary RPC.
+func (c *Client) unaryRPC(ctx context.Context, cfg *RPCConfig, f func(context.Context) error) error {
```

Comment:
Will this eventually be moved to a separate package so `client` can also use it?

Reply:
Maybe? I thought of this too, but it is a bit hard for me to tell right now because this is specific to `RPCConfig` which is a bit specific CAS (it has `MaxItems` for example).

In Go's author own words, "little copying is better than little dependency" https://youtu.be/PAAkCSZUG1c?t=567
We are trying to save 5 lines of code here, at least for now.